server/client.go

```diff
@@ -2491,47 +2492,86 @@ func (c *client) addShadowSubscriptions(acc *Account, sub *subscription) error {
 	}
 
 	var (
-		_ims   [16]ime
-		ims    = _ims[:0]
-		tokens []string
-		tsa    [32]string
-		hasWC  bool
+		_ims           [16]ime
+		ims            = _ims[:0]
+		imTsa          [32]string
+		tokens         []string
+		tsa            [32]string
+		hasWC          bool
+		tokensModified bool
 	)
 
 	acc.mu.RLock()
-	// Loop over the import subjects. We have 3 scenarios. If we have an
+	subj := string(sub.subject)
+	if len(acc.imports.streams) > 0 {
+		tokens = tsa[:0]
+		tokenizeSubjectIntoSlice(subj, &tokens)
+		for _, tk := range tokens {
+			if tk == pwcs {
+				hasWC = true
+			}
```

Comment:
I think that a `break` makes sense here: as soon as you discover a `*` token, you know there is wildcard.

Reply:
agreed